src/nvim/api/vim.c

```diff
@@ -2250,6 +2250,56 @@ Array nvim_get_proc_children(Integer pid, Error *err)
   return rvobj;
 }
 
+Boolean nvim_add_autocommand(
+    uint64_t channel_id,
+    String event, String pat, DictionaryOf(LuaRef) opts, Error *err
+) FUNC_API_SINCE(6)
+{
+  char_u cmd = NUL;
+  char_u *next_ev;
+  LuaRef cb = LUA_NOREF;
+
+  event_T event_nr = event_name2nr((char_u *)event.data, &next_ev);
```

Comment:
`next_ev` is only needed for `event_name2nr` but not used here. Maybe we can have better API for converting names to `event_T` enum.